### DISCUSSION TITLE
Clarification on when a transition will actually happen?

### DISCUSSION THREAD

**Discussion by auralluring** [imjp94/gd-YAFSM#34] [Q&A]
So I have a state machine set up, and for the most part, it's pretty helpful. But for the past couple hours I've been very frustrated and can't help but feel like I'm missing something. 
This is the state machine in action (it's nested)
![Screenshot from 2021-11-24 16-33-33](https://user-images.githubusercontent.com/45722265/143326100-e6b6c35c-0c9a-4194-b493-e684e1fb6f14.png)
What I don't understand is even though the condition for that transition is true, it doesn't go through, no matter how long I wait. Is there some other, hidden condition that a transition has to meet before it does it? Or is it a bug? Any insight would be helpful, thanks in advance.

**Comment by imjp94** [comment]
I guess the problem lies with the DangerChecking's 2 transitions checking on the same Boolean(can_see_player), which means DangerChecking will never stay for more than 1 frame. 

And that is way too fast for the debugger to visualize. So what you see on the screen is that it always stays in Run, but it is actually in a loop.

To prove my point, you can try capturing the `transited` signal and print out the transition. Or simply just set can_see_player to false.

If the problem persists, do share the output you get from capturing `transited` signal
